src/events/neo4j.js

```diff
@@ -0,0 +1,386 @@
+        onError(err) {
+            eventInterface.setException(dbApiEvent, err);
+            // eslint-disable-next-line prefer-rest-params
+            return observer.onError.apply(session, sessionArguments);
+        },
+    };
+}
+
+
+/**
+ * Create new serverlessEvent for Neo4J run event (any type of run) and
+ *  add relevant metadata
+ * @param {neo4j.Session|neo4j.Transaction} session A DB session (could be Session of Transaction)
+ * @param {number} startTime The Event start time
+ * @returns {serverlessEvent.Event} The new Event
+ */
+function createNewNeo4jEvent(session, startTime) {
+    const { host, port } = getAddressInfo(session);
+
+    const {
+        metadata, fullMetadata,
+    } = getNeo4JSessionMetadata(session);
+
+    metadata.port = port;
+
+    const resource = new serverlessEvent.Resource([
+        host,
+        'neo4j',
+        metadata.operation,
+    ]);
+
+    const dbApiEvent = new serverlessEvent.Event([
+        `neo4j-${uuid4()}`,
+        utils.createTimestampFromTime(startTime),
+        null,
+        'neo4j',
+        0,
+        errorCode.ErrorCode.OK,
+    ]);
+
+    dbApiEvent.setResource(resource);
+
+    eventInterface.addToMetadata(dbApiEvent, metadata, fullMetadata);
+
+    return dbApiEvent;
+}
+
+
+/**
+ * Wraps neo4j.Session.run functions with tracing
+ * @param {Function} wrappedFunction The function to wrap from Neo4j
+ * @returns {Function} The wrapped function
+ */
+function neo4jSessionRunWrapper(wrappedFunction) {
+    return function internalNeo4jSessionRunWrapper(...args) {
+        const relevantArgs = getArgsFromFunction(...args, this.constructor.name);
+        const {
+            query, params, transactionConfig,
+        } = relevantArgs;
+
+        utils.debugLog('User called Neo4j wrapped Session run function');
+
+        try {
+            const startTime = Date.now();
+
+            const dbApiEvent = createNewNeo4jEvent(this, startTime);
+
+            eventInterface.addToMetadata(
+                dbApiEvent,
+                {},
+                { query, param: params, transaction_config: transactionConfig }
+            );
+
+
+            const resultResponse = wrappedFunction.apply(
+                this,
+                [query, params, transactionConfig]
+            );
+
+            const originalSubscribe = resultResponse.subscribe;
+
+            // Override the Result subscribe with patched subscriber
+            // eslint-disable-next-line func-names
+            resultResponse.subscribe = function (observer) {
+                originalSubscribe.call(this, getPatchedSubscribeMethods(
+                    this,
+                    observer,
+                    arguments,
+                    dbApiEvent,
+                    startTime
+                ));
+            };
+
+            return resultResponse;
+        } catch (error) {
+            tracer.addException(error);
+            return wrappedFunction.apply(this, [query, params, transactionConfig]);
```

Comment:
same as below

Reply:
Fixed by moving the apply out of the try scope, i think that will solve the issue, right?